prestoadmin/prestoclient.py

```diff
@@ -126,19 +145,28 @@ def get_response_from(self, uri):
         Sends a GET request to the Presto server at the specified next_uri
         and updates the response
         """
-        try:
-            conn = urlopen(uri, None, URL_TIMEOUT_MS)
-            answer = conn.read()
-            conn.close()
+        parts = list(urlparse.urlsplit(uri))
```

Comment:
What is this bit doing exactly? Can you make it more explicit/less hacky looking?